--- a/configure.ac
+++ b/configure.ac
@@ -271,5 +271,6 @@ else
   dnl libgstlibav.la: libs to statically link to        
-  LIBAV_LIBS="\$(top_builddir)/gst-libs/ext/libav/libavformat/libavformat.a \
-               \$(top_builddir)/gst-libs/ext/libav/libavcodec/libavcodec.a \
-               \$(top_builddir)/gst-libs/ext/libav/libavutil/libavutil.a"
+  LIBAV_LIBS="-Wl,\$(top_builddir)/gst-libs/ext/libav/libavformat/libavformat.a \
+              -Wl,\$(top_builddir)/gst-libs/ext/libav/libavcodec/libavcodec.a \
+              -Wl,\$(top_builddir)/gst-libs/ext/libav/libavutil/libavutil.a \
+              -Wl,-lbz2,-lz"
 
@@ -281,4 +282,4 @@ else
   dnl libgstswscale.la: libs to statically link to
-  SWSCALE_LIBS="\$(top_builddir)/gst-libs/ext/libav/libswscale/libswscale.a \
-                \$(top_builddir)/gst-libs/ext/libav/libavutil/libavutil.a"
+  SWSCALE_LIBS="-Wl,\$(top_builddir)/gst-libs/ext/libav/libswscale/libswscale.a \
+                -Wl,\$(top_builddir)/gst-libs/ext/libav/libavutil/libavutil.a"
 
--- a/ext/libav/Makefile.am
+++ b/ext/libav/Makefile.am
@@ -31,10 +31,6 @@ libgstlibav_la_LIBADD = $(FFMPEG_LIBS)
 libgstlibav_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS) $(DARWIN_LDFLAGS)
 libgstlibav_la_LIBTOOLFLAGS = --tag=disable-static
 
-if HAVE_LIBAV_UNINSTALLED
-libgstlibav_la_DEPENDENCIES = $(LIBAV_LIBS)
-endif
-
 
 noinst_HEADERS = \
 	gstav.h \
--- a/ext/libswscale/Makefile.am
+++ b/ext/libswscale/Makefile.am
@@ -9,7 +9,3 @@ libgstlibavscale_la_LIBADD = $(SWSCALE_
 	$(GST_BASE_LIBS) $(ORC_LIBS) $(LIBM) -lz
 libgstavscale_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS) $(DARWIN_LDFLAGS)
 libgstavscale_la_LIBTOOLFLAGS = --tag=disable-static
-
-if HAVE_LIBAV_UNINSTALLED
-libgstavscale_la_DEPENDENCIES = $(SWSCALE_LIBS)
-endif
