Causes compile failure on strptime call

--- gstreamer1.0-plugins-libav-1.2.3-1.x86_64/origsrc/gst-libav-1.2.3/gst-libs/ext/libav/configure	2014-02-08 05:57:18.000000000 -0600
+++ gstreamer1.0-plugins-libav-1.2.3-1.x86_64/src/gst-libav-1.2.3/gst-libs/ext/libav/configure	2014-03-14 12:40:19.002375200 -0500
@@ -2783,7 +2783,7 @@ if test "$?" != 0; then
 fi
 
 add_cppflags -D_ISOC99_SOURCE
-check_cflags -std=c99
+check_cflags -std=gnu99
 check_cc -D_FILE_OFFSET_BITS=64 <<EOF && add_cppflags -D_FILE_OFFSET_BITS=64
 #include <stdlib.h>
 EOF
@@ -2791,7 +2791,7 @@ check_cc -D_LARGEFILE_SOURCE <<EOF && ad
 #include <stdlib.h>
 EOF
 
-check_host_cflags -std=c99
+check_host_cflags -std=gnu99
 check_host_cflags -Wall
 
 check_64bit(){
